pkg/ring/token_checker.go

```diff
@@ -0,0 +1,253 @@
+package ring
+
+import (
+	"context"
+	"flag"
+	"sync"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+)
+
+// TokenCheckerConfig is the config to configure a TokenChecker.
+type TokenCheckerConfig struct {
+	// Used by ingester when set to true.
+	CheckOnCreate   bool `yaml:"check_on_create,omitempty"`
+	CheckOnAppend   bool `yaml:"check_on_append,omitempty"`
+	CheckOnTransfer bool `yaml:"check_on_transfer,omitempty"`
```

Comment:
That's a fair point. I think I mostly agree, although it's somewhat nice to group them logically since they are all related. 

I'll move them to the ingester.